ballerina/msg_listener.bal

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.runtime;
+import ballerina/log;
+import ballerina/task;
+
+# Represents the default error type.
+public type Error distinct error;
+
+# Represents the message store listener configuration,
+public type StoreListenerConfiguration record {|
+    # The interval in seconds at which the listener polls for new messages
+    decimal pollingInterval = 1;
+    # The maximum number of retries for processing a message. 
+    # If set to 0, the message will not be retried
+    int maxRetries = 3;
```

Comment:
```suggestion
    `int:Unsigned8` maxRetries = 3;
```

Reply:
Lang issue: https://github.com/ballerina-platform/ballerina-lang/issues/44084

Reply:
+1 for using `int` as the type. Once the lang issue is fixed we could move the implementation to use `int:Unsigned8` type.